### PR TITLE
User id backend frontend

### PR DESCRIPTION
**Pull request by XinranTaoAngela**
enabled frontend to display userID from protected route

### CODE REVIEW

File: back-end/authRoutes.js [hunk 1]
```diff
@@ -92,10 +92,26 @@ const authenticateToken = (req, res, next) => {
...
+    // Retrieve user details from the database using the userId from req.user
+    const user = await User.findById(req.user.userId);
+
+    // Check if user exists
+    if (!user) {
+      return res.status(404).json({ message: 'User not found' });
+    }
+
+    // Respond with user details
+    res.json({
```
Comment (SivanC) [inline back-end/authRoutes.js:107]: add 200 status

File: front-end/src/Header.js
Comment (SivanC) [inline front-end/src/Header.js:1]: just remember to restore this file with `git restore Header.js` when merging

File: back-end/authRoutes.js
Comment (SivanC) [inline back-end/authRoutes.js:1]: unsure what all the changes in the login route are, and lots of semicolons have been added

File: back-end/authRoutes.js [hunk 2]
```diff
@@ -80,22 +82,41 @@ router.post('/logout', (req, res) => {
...
+      message: `Hello ${user.first_name} ${user.last_name}`,
```
Comment (SivanC) [inline back-end/authRoutes.js:113]: purpose of the message?

File: back-end/test/authroutes.test.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ const sinon = require('sinon');
 const bcrypt = require('bcrypt');
 const jwt = require('jsonwebtoken'); // Import jsonwebtoken
 const mongoose = require('mongoose');
-const app = require('../app'); // Import your Express app (not server)
-const { User } = require('../models/User'); // Adjust this to the path of your User model
+const { app} = require('../app');
```
Comment (SivanC) [inline back-end/test/authroutes.test.js:7]: inconsistent spacing, and no need for brackets if its the only export

File: front-end/src/Header.css
Comment (SivanC) [inline front-end/src/Header.css:1]: restore this too

File: back-end/test/authroutes.test.js [hunk 4]
```diff
@@ -26,55 +26,55 @@ describe('User Routes', () => {
         last_name: 'Doe'
       });
 
-      const res = await chai.request(app) // Use your Express app here
-        .post('/api/auth/signup') // Make sure the route path is correct
+      chai.request(app)
+        .post('/api/signup') // Adjust the route path based on your Express routes
```
Comment (SivanC) [inline back-end/test/authroutes.test.js:30]: the route is just /signup right now

**Review by SivanC — CHANGES_REQUESTED** [review]
Some unnecessary changes or adding semicolons that don't need to be added, also tests don't pass, but the route itself looks fine.

File: back-end/authRoutes.js (hunk 2 above)
Comment (XinranTaoAngela) [reply back-end/authRoutes.js:113]: deleted this message

**Review by SivanC — APPROVED** [review]

**Pull request merged**